test/es-module/test-esm-loader-resolve-type.mjs

```diff
@@ -0,0 +1,42 @@
+// Flags: --loader ./test/fixtures/es-module-loaders/hook-resolve-type.mjs
+import { allowGlobals } from '../common/index.mjs';
+import * as fixtures from '../common/fixtures.mjs';
+import { strict as assert } from 'assert';
+import * as fs from 'fs';
+import { fileURLToPath, URL } from 'url';
```

Comment:
That's not necessary, `URL` is available on the global scope.
```suggestion
```

Reply:
removed URL from that import, `fileURLToPath` is still needed because of the exception with `rmSync(URL)`